pkg/result/processors/fixer.go

```diff
@@ -43,218 +53,221 @@ func (p Fixer) Process(issues []result.Issue) ([]result.Issue, error) {
+		var edits []diff.Edit
+		for linter := range linterToEdits {
+			if _, found := excludedLinters[linter]; !found {
+				edits = append(edits, linterToEdits[linter]...)
+			}
+		}
 
-	if len(lineIssues) == 1 && lineIssues[0].Replacement.Inline == nil {
-		return &lineIssues[0]
+		editsByPath[path], _ = validateEdits(edits) // remove duplicates. already validated.
 	}
 
-	// check issues first
-	for ind := range lineIssues {
-		li := &lineIssues[ind]
+	var editError error
 
-		if li.LineRange != nil {
-			p.log.Infof("Line %d has multiple issues but at least one of them is ranged: %#v", lineNum, lineIssues)
-			return &lineIssues[0]
+	// Now we've got a set of valid edits for each file. Apply them.
+	for path, edits := range editsByPath {
+		contents, err := p.fileCache.GetFileBytes(path)
+		if err != nil {
+			editError = errors.Join(editError, err)
+			continue
 		}
 
-		inline := li.Replacement.Inline
+		out, err := diff.ApplyBytes(contents, edits)
+		if err != nil {
+			editError = errors.Join(editError, err)
+			continue
+		}
 
-		if inline == nil || len(li.Replacement.NewLines) != 0 || li.Replacement.NeedOnlyDelete {
-			p.log.Infof("Line %d has multiple issues but at least one of them isn't inline: %#v", lineNum, lineIssues)
-			return li
+		// Try to format the file.
+		if formatted, err := format.Source(out); err == nil {
+			out = formatted
 		}
 
-		if inline.StartCol < 0 || inline.Length <= 0 || inline.StartCol+inline.Length > len(origLine) {
-			p.log.Warnf("Line %d (%q) has invalid inline fix: %#v, %#v", lineNum, origLine, li, inline)
-			return nil
+		if err := os.WriteFile(path, out, filePerm); err != nil {
+			editError = errors.Join(editError, err)
+			continue
 		}
 	}
 
-	return p.applyInlineFixes(lineIssues, origLine, lineNum)
+	return notFixableIssues, editError
 }
 
-func (p Fixer) applyInlineFixes(lineIssues []result.Issue, origLine []byte, lineNum int) *result.Issue {
-	sort.Slice(lineIssues, func(i, j int) bool {
-		return lineIssues[i].Replacement.Inline.StartCol < lineIssues[j].Replacement.Inline.StartCol
-	})
-
-	var newLineBuf bytes.Buffer
-	newLineBuf.Grow(len(origLine))
-
-	//nolint:misspell // misspelling is intentional
-	// example: origLine="it's becouse of them", StartCol=5, Length=7, NewString="because"
+func (Fixer) Finish() {}
 
-	curOrigLinePos := 0
-	for i := range lineIssues {
-		fix := lineIssues[i].Replacement.Inline
-		if fix.StartCol < curOrigLinePos {
-			p.log.Warnf("Line %d has multiple intersecting issues: %#v", lineNum, lineIssues)
-			return nil
-		}
+func (p Fixer) printStat() {
+	p.sw.PrintStages()
+}
 
-		if curOrigLinePos != fix.StartCol {
-			newLineBuf.Write(origLine[curOrigLinePos:fix.StartCol])
+func skipTextEditWithoutPosition(issue *result.Issue) bool {
+	var onlyMessage int
+	var count int
+	for _, sf := range issue.SuggestedFixes {
+		for _, edit := range sf.TextEdits {
+			count++
+			if edit.Pos == token.NoPos && edit.End == token.NoPos {
```

Comment:
TIL! Which linters uses these suggested fixes? Or is it just proactively if one ever does?

Reply:
There is one linter that uses suggestions without a position, this is not a proactive protection.